client/app/pages/dashboards/ShareDashboardDialog.jsx

```diff
@@ -104,10 +104,10 @@ class ShareDashboardDialog extends React.Component {
         footer={null}
       >
         <Form layout="horizontal">
-          {this.props.hasQueryParams && (
+          {!this.props.hasOnlySafeQueries && (
             <Form.Item>
               <Alert
-                message="Sharing is currently not supported for dashboards containing queries with parameters."
+                message="For your security, sharing is currently not supported for dashboards containing queries with text parameters. Consider changing the text parameters in your query to a different type."
```

Comment:
This message shouldn't be visible if dashboard does not have parameters at all.

Reply:
If a dashboard doesn't have parameters at all - that means all its queries are safe, so `hasOnlySafeQueries` is true and `!hasOnlySafeQueries` is false, so it won't show.